neqo-http3/src/connection_server.rs

```diff
@@ -193,13 +193,20 @@ impl Http3ServerHandler {
                         HFrame::PriorityUpdatePush { element_id, priority } => {
                             // TODO: check if the element_id references a promised push stream or
                             //       is greater than the maximum Push ID.
-
                             self.events.priority_update(element_id, priority);
                             Ok(())
                         }
                         HFrame::PriorityUpdateRequest { element_id, priority } => {
-                            // TODO: check that the element_id references a request stream
-                            //       within the client-sided bidirectional stream limit
+                            // check that the element_id references a request stream
+                            // within the client-sided bidirectional stream limit
+                            let element_stream_id = StreamId::new(element_id);
+                            if !element_stream_id.is_bidi()
+                                || !element_stream_id.is_client_initiated()
+                                || !conn.is_in_remote_bidi_limit(element_stream_id)
```

Comment:
Can you write a test for this?
You can copy his one: https://github.com/mozilla/neqo/blob/37bb62fa7b43ac624406090ba4f94f4cc082790b/neqo-http3/src/server.rs#L560
and send PRIORITY_UPDATE with a wrong ID.